generator/src/main/java/com/scottlogic/deg/generator/decisiontree/MaxStringLengthInjectingDecisionTreeFactory.java

```diff
@@ -0,0 +1,64 @@
+package com.scottlogic.deg.generator.decisiontree;
+
+import com.google.inject.Inject;
+import com.scottlogic.deg.generator.Profile;
+import com.scottlogic.deg.generator.constraints.atomic.IsStringShorterThanConstraint;
+import com.scottlogic.deg.generator.generation.GenerationConfig;
+import com.scottlogic.deg.generator.inputs.RuleInformation;
+import com.scottlogic.deg.schemas.v0_1.RuleDTO;
+
+import java.util.Collections;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Decorator over a DecisionTreeFactory to inject a &lt;shorterThan X&gt; constraint at the root node for every field
+ */
+public class MaxStringLengthInjectingDecisionTreeFactory implements DecisionTreeFactory{
+    private final int maxLength;
+    private final DecisionTreeFactory underlyingFactory;
+
+    @Inject
+    public MaxStringLengthInjectingDecisionTreeFactory(ProfileDecisionTreeFactory underlyingFactory) {
+        this(underlyingFactory, GenerationConfig.Constants.MAX_STRING_LENGTH);
+    }
+
+    public MaxStringLengthInjectingDecisionTreeFactory(DecisionTreeFactory underlyingFactory, int maxLength) {
+        this.underlyingFactory = underlyingFactory;
+        this.maxLength = maxLength;
+    }
+
+    @Override
+    public DecisionTreeCollection analyse(Profile profile) {
+        DecisionTreeCollection collection = underlyingFactory.analyse(profile);
+        return new DecisionTreeCollection(
+            collection.getFields(),
+            collection.getDecisionTrees()
+                .stream()
+                .map(this::injectConstraints)
+                .collect(Collectors.toList())
+        );
+    }
+
+    private DecisionTree injectConstraints(DecisionTree tree) {
+        Set<RuleInformation> rules = Collections.singleton(createRule());
+
+        return new DecisionTree(
+            tree.rootNode.addAtomicConstraints(
```

Comment:
will this add duplicates to the tree if the same field is representative over multiple rules?

Reply:
It shouldn't as all the rules have been merged into one tree come this point

Reply:
this is being called during analyse, not getMergedProfile, it's still multiple tree's at this point.

Reply:
I see, yes it would, then they would be merged into one instance as part of the `merge()` method; not ideal.

Reply:
This should now be sorted.

~`public DecisionTreeCollection DecisionTreeFactory.analyse(Profile)`~

is now

`public DecisionTree DecisionTreeFactory.analyse(Profile)`

Reply:
that seems better.

more minor, would it be worth renaming analyse to createTree